docs/examples.rst

```diff
@@ -362,3 +362,82 @@ Calculate the token count from the token balance of an account address.
 Balance is *always* an integer in the currency's smallest natural unit (equivalent to 'wei' for ether). Token balance is typically used only for backend calculations.
 
 Token count *may* be a integer or fraction in the currency's primary unit (equivalent to 'eth' for ether). Token count is typically displayed to the user on the front-end since it is more readable.
+
+
+Creation of the contract factory from the ERC20 ABI
+---------------------------------------------------
+Using this command, you can create a contract factory from the ERC20 ABI. You need to specify the contract address
+
+.. testsetup::
+    import os
+    os.environ['INFURA_API_KEY'] = "b2679bb624354d1b9a2586154651b51f"
```

Comment:
A step in the right direction, but I'd like to push this a little further.

Right now, as I understand it, some of the examples below are reading token balances from the live [Unicorn Token contract](https://etherscan.io/address/0x89205A3A3b2A69De6Dbf7f01ED13B2108B2c43e7).  This is a problem because our tests can/will fail when either the HTTP requests to infura fail, infura's API changes, this API key is revoked, or the data in the contract itself changes.

To fix this, we need to update this code to instead only read data from a local in-memory EVM, aka, the `EthereumTesterProvider`.

```
.. testsetup::

    from web3 import Web3
    w3 = Web3(Web3.EthereumTesterProvider())
    bytecode = "<the-deployment-bytecode-for-a-standard-ERC20-contract>"
    abi = "<the-ERC20-ABI>"
    factory = w3.eth.contract(abi=abi, bytecode=bytecode)
    deploy_txn = factory.constructor(...).transact()
    deploy_receipt = w3.eth.waitForTransactionReceipt(deploy_txn)
    contract_address = deploy_receipt['contractAddress']
```

This will give you an in-memory ERC20 token contract to work with.  You probably need to add a few extra bits to populate some account balances for use in the examples below.  Then, in the examples below, you can just use that contract address and only use the local.

Please take a look at the doctest API here: http://www.sphinx-doc.org/en/master/usage/extensions/doctest.html

These directives have a `[group]` parameter that can be used to allow you to apply a single `testsetup` block to multiple doctests.

Reply:
@pipermerriam the EthereumTesterProvider and EthereumTester methods don't populate accounts for use within the blocks. And ideas?